src/step.rs

```diff
@@ -96,57 +100,73 @@ where
     F: Fn(&InterpCx) -> ShouldContinue,
 {
     let mut message = None;
-    loop {
-        if pcx.ecx.stack().len() <= 1 && is_ret(&pcx.ecx) {
-            break;
-        }
-        match pcx.ecx.step() {
-            Ok(true) => {
-                *pcx.step_count += 1;
-                crate::watch::step_callback(pcx);
-
-                if let Some(frame) = pcx.ecx.stack().last() {
-                    if let Some(block) = frame.block {
-                        let blck = &frame.body.basic_blocks()[block];
-                        if frame.stmt != blck.statements.len()
-                            && crate::should_hide_stmt(&blck.statements[frame.stmt])
-                            && !pcx.config.bptree.is_at_breakpoint(&pcx.ecx)
-                        {
-                            continue;
-                        }
-                    } else {
-                        // Unwinding, but no cleanup for this frame
-                        // FIXME make step behaviour configurable
-                    }
+    let ret: InterpResult<'_, _> = (|| {
```

Comment:
You could use a `try` block.

Reply:
Do you mean this: https://doc.rust-lang.org/nightly/unstable-book/language-features/try-blocks.html? Or do you have some other feature in mind?

Reply:
Yes, that feature. It is used elsewhere in priroda too.

Reply:
Thanks! I never used this feature before.